pkg/users/users.js

```diff
@@ -165,11 +165,31 @@ async function getLogins() {
         console.warn("Unexpected error when getting logged in accounts", err);
     }
 
+    // shadow-utils passwd supports an --all flag which is lacking on RHEL and
+    // stable Fedora releases. Available at least on Fedora since
+    // shadow-utils-4.14.0-5.fc40 (currently known as rawhide).
+    const locked_users_map = {};
+    try {
+        const locked_statuses = await cockpit.spawn(["passwd", "-S", "--all"], { superuser: "require", environ: ["LC_ALL=C"] });
+        // Slice off the last empty line
+        for (const line of locked_statuses.trim().split('\n')) {
+            const username = line.split(" ")[0];
+            const status = line.split(" ")[1];
+            locked_users_map[username] = status == "L";
+        }
+    } catch (err) {
+        // Only warn when it is unrelated to --all.
+        if (err.exit_status !== 253) {
```

Comment:
... then you can check it here and check the message, see https://cockpit-project.org/guide/latest/cockpit-spawn.html#cockpit-spawn-catch